agnpy/tests/test_sedintegrable.py

```diff
@@ -0,0 +1,82 @@
+import numpy as np
+import math
+import astropy.units as u
+from astropy.constants import m_e, h
+from astropy.coordinates import Distance
+from agnpy.spectra import PowerLaw
+from agnpy.emission_regions import Blob
+from agnpy.synchrotron import Synchrotron
+from ..utils.sedintegrable import SedFluxIntegrable
+
+
+def _estimate_powerlaw_integral(x_values, y_values):
+    slope, intercept = np.polyfit(np.log10(x_values.value), np.log10(y_values.value), 1)
+    power = slope + 1
+    estimated_integral = (10 ** intercept) * (
+            x_values[-1].value ** power / power - x_values[0].value ** power / power)
+    return estimated_integral
+
+
+def _sample_synchrotron_model():
+    # Based on the synchrotron example from agnpy documentation
+    R_b = 1e16 * u.cm
+    n_e = PowerLaw.from_total_energy(1e48 * u.erg, 4 / 3 * np.pi * R_b ** 3, p=2.8, gamma_min=1e2,
+                                     gamma_max=1e7, mass=m_e)
+    blob = Blob(R_b, z=Distance(1e27, unit=u.cm).z, delta_D=10, Gamma=10, B=1 * u.G, n_e=n_e)
+    synch = Synchrotron(blob)
+    return synch
+
+
+class TestSedIntegrable:
+
+    def test_flat_integral(self):
+        """Integrate over flat SED (Fnu equal to 1.0 for all frequencies)"""
+        class FlatSedGenerator(SedFluxIntegrable):
+            def sed_flux(self, nu):
+                return np.full(fill_value=1.0, shape=nu.shape, dtype=np.float64) * (u.erg / (u.cm ** 2 * u.s * u.Hz)) * nu
+
+        nu_min = 10 * u.Hz
+        nu_max = 10 ** 20 * u.Hz
+        flux = FlatSedGenerator().integrate_flux(nu_min, nu_max)
+        assert flux == 1e+20 * u.erg / (u.cm ** 2 * u.s)
+
+
+    def test_synchrotron_energy_flux_integral(self):
+        """Integrate over sample synchrotron flux and compare with the value estimated using the power law integral"""
+        synch = _sample_synchrotron_model()
+
+        nu_min_log = 13
+        nu_max_log = 20
+        nu_min = 10 ** nu_min_log * u.Hz
+        nu_max = 10 ** nu_max_log * u.Hz
+
+        nu = np.logspace(nu_min_log, nu_max_log) * u.Hz
+        Fnu = synch.sed_flux(nu) / nu
+        estimated_integral = _estimate_powerlaw_integral(nu, Fnu)
```

Comment:
Sorry if I am a bit lost. You are fitting the synchrotron spectrum with a power law and integrating the result, right? And this is your comparison term for the test. I assume you are integrating the synchrotron spectrum then in a region of frequency in which it is a straight power-law, is that right?

Reply:
Exactly. I took the synchrotron example from agnpy tutorials, only cut down the SED to the range when it fits to the power-law.
Then I calculate integral analytically for this range, and compare with the results calculcated by the integrate_flux method